src/librustc_target/spec/apple_base.rs

```diff
@@ -51,3 +51,19 @@ pub fn macos_llvm_target(arch: &str) -> String {
     let (major, minor) = macos_deployment_target();
     format!("{}-apple-macosx{}.{}.0", arch, major, minor)
 }
+
+pub fn macos_link_env() -> Vec<(String, String)> {
```

Comment:
This seems a bit of an odd interface since it actually only returns env vars which should be cleared out, not env vars which should be set. Do we know what the various tools will behave right when set to blank values which effectively disables them? The alternative would be to extend `link_env` with the ability to delete env vars as well (e.g. `env_remove`).

Additionally, it seems like this would also want to encompass the ios targets perhaps? For example if you set `SDKROOT` for osx or `MACOSX_DEPLOYMENT_TARGET`, should those values be automatically cleared out for ios compliation in addition to macos compilation?

Reply:
> This seems a bit of an odd interface since it actually only returns env vars which should be cleared out, not env vars which should be set. Do we know what the various tools will behave right when set to blank values which effectively disables them? The alternative would be to extend `link_env` with the ability to delete env vars as well (e.g. `env_remove`).

If the variable is set to an empty string, the linker behaves as if the environment variable wasn't set. We can rework this to remove environment variables instead if that feels better.

> Additionally, it seems like this would also want to encompass the ios targets perhaps? For example if you set `SDKROOT` for osx or `MACOSX_DEPLOYMENT_TARGET`, should those values be automatically cleared out for ios compliation in addition to macos compilation?

Yes, this makes sense. We should clear `SDKROOT` if it's set to the wrong platform when targeting iOS. The value of `MACOSX_DEPLOYMENT_TARGET` doesn't appear to be meaningful in the same way that `IPHONEOS_DEPLOYMENT_TARGET` is (i.e. it doesn't appear to be used by the linker to infer the target platform), so perhaps we don't need to clear this one.